src/components/billingRecord/IFXBillingRecordList.vue

```diff
@@ -292,20 +295,24 @@ export default {
           missing.push(i)
         }
       }
-      Promise.all(missing.map((i) => me.getFullBillingRecordByItemIndex(i)))
-        .then(() => {
-          const toBeUpdated = []
-          me.items.forEach((it1) => {
-            items.forEach((it2) => {
-              if (it1.id === it2.id) {
-                it1.billingRecordStates.push({ name: state, user: '', approvers: [], comment: '' })
-                toBeUpdated.push(it1)
-              }
-            })
+      const promises = missing.map((i) => me.getFullBillingRecordByItemIndex(i))
+      while (promises.length) {
+        await new Promise(r => setTimeout(r, 500));
+        await Promise.all(promises.splice(0, this.promiseBatchSize))
+          .catch((err) => {
+            console.log('error from one of the promises ', err)
           })
-          console.log(toBeUpdated)
-          return this.$api.billingRecord.bulkUpdate(toBeUpdated, this.facility.applicationUsername)
+      }
+      const toBeUpdated = []
+      this.items.forEach((it1) => {
```

Comment:
You might want to structure this differently to allow the loop to be broken out of once a match is found.
```suggestion
      items.forEach((it2) => {
        for (i = 0; i < this.items.length; i++) {
          if (this.items[i].id === it2.id) {
              const it1 = this.items[i]
              it1.billingRecordStates.push({ name: state, user: '', approvers: [], comment: '' })
             toBeUpdated.push(it1)
             break;
           }
         }
      }
```
Or something like that. The point is that `forEach` doesn't allow one to use `break`.